### ISSUE TITLE
Grafana alert group or folder change creates a Postgresql error

### ISSUE THREAD

**Issue by saimon46** [grafana/grafana#55901]
**What happened**:
Changing the grafana alert group or folder triggers a Postgresql DB error in a SELECT query that runs when saving the alert.

**What you expected to happen**:
The alert should be saved with the details saved without any issue.

**How to reproduce it (as minimally and precisely as possible)**:
1. Use Grafana with Postgresql backend.
2. Create an alert and save it.
3. Modify the alert changing the alert group or the alert folder.
4. Save the alert, it fails. With this error:
![Screenshot 2022-09-28 at 11 27 02](https://user-images.githubusercontent.com/6933309/192744336-9307f550-334b-45a0-a346-6955665a5241.png)

**Anything else we need to know?**:
We identified the issue debugging the SQL query that runs against the Postgresql DB.
```
SELECT A.* FROM "alert_rule" AS "A" INNER JOIN alert_rule AS B ON A.org_id = B.org_id AND A.namespace_uid = B.namespace_uid AND A.rule_group = B.rule_group AND B.uid = $1 WHERE (A.org_id = $2)
```
This SQL query fails because the alias `"A"` is defined with quotes but then used without quotes in the rest of the query.
The error on the DB is the following:
```
2022-09-28 08:12:43.327 UTC [4468] ERROR: missing FROM-clause entry for table "a" at character 67
2022-09-28 08:12:43.327 UTC [4468] STATEMENT: SELECT A.* FROM "alert_rule" AS "A" INNER JOIN alert_rule AS B ON A.org_id = B.org_id AND A.namespace_uid = B.namespace_uid AND A.rule_group = B.rule_group AND B.uid = $1 WHERE (A.org_id = $2)
```
The code that runs the failed query is this: https://github.com/grafana/grafana/blob/v9.1.6/pkg/services/ngalert/store/alert_rule.go#L143.
Probably this is working on Mysql but not in Postgresql.

**Possible resolution**:
A resolution is to remove the quotes from the Alias but not easy since it's a function that generates it and I don't know the compatibility with Mysql or other backends.
```
		err := sess.Table("alert_rule").Alias("A").Join(
			"INNER",
			"alert_rule AS B", "A.org_id = B.org_id AND A.namespace_uid = B.namespace_uid AND A.rule_group = B.rule_group AND B.uid = ?", query.UID,
		).Where("A.org_id = ?", query.OrgID).Select("A.*").Find(&result)
```
Probably the best would be to add all the quotes in all the other `A` calls. Not sure, I let you find the best way to do it. 

**Environment**:
- Grafana version: 9.1.6, but it fails in 9.1.3 too, with Postgresql 12 backend.
- Data source type & version: All of them, issue not related to datasources
- OS Grafana is installed on: Rocky Linux 8.6
- User OS & Browser: Chrome on MacOS
- Grafana plugins: Clickhouse plugin
- Others: None

**Comment by OscarStockill** [comment]
I also get this error (or something very similiar) when trying to update rules via the grafana ruler api.

Grafana version: 9.1.5
Postgresql

Hitting the post endpoint at `/api/ruler/grafana/api/v1/rules/<data_source>`
`"message":"failed to update rule group: failed to query database for a group of alert rules: pq: missing FROM-clause entry for table \"a\""` 

This was while trying to use the api to bulk remove labels from alerts.

I have also seen the exact error saimon46 explains editing alerts in the UI.

**Comment by milinpatel13298** [comment]
I am also getting a similar error for which I had created a [topic](https://community.grafana.com/t/error-while-creating-alerts/73633) in the forum as well
`Failed to save rule: failed to update rule group: failed to query database for a group of alert rules: pq: missing FROM-clause entry for table "a"`
I'd also like to understand how/where these groups are pre-defined as mentioned in the [documentation](https://grafana.com/docs/grafana/latest/alerting/alerting-rules/create-grafana-managed-rule/). Particularly, because I was able to enter "server" as a valid value in my other alerts. It was only when I needed to enter a different group value that I discovered this issue.
